test-browser/tests/solidityUnittests.js

```diff
@@ -59,34 +62,62 @@ module.exports = {
-    .pause(10000)
+    .pause(15000)
     .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', 'browser/ks2b_test.sol (kickstarterTest)')
     .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', '✓ (Check project exists)')
     .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', '✓ (Check project is fundable)')
   },
 
-  'Should run all test files': function (browser) {
+  'Should fail on compilation': function (browser) {
     browser.waitForElementPresent('*[data-id="verticalIconsKindfileExplorers"]')
+    .addFile('compilationError_test.sol', sources[0]['browser/compilationError_test.sol'])
     .clickLaunchIcon('fileExplorers')
-    .switchFile('browser/3_Ballot.sol')
+    .switchFile('browser/compilationError_test.sol')
     .clickLaunchIcon('solidityUnitTesting')
+    .click('*[data-id="testTabCheckAllTests"]')
+    .click('.singleTestLabel:nth-of-type(4)')
     .scrollAndClick('*[data-id="testTabRunTestsTabRunAction"]')
-    .pause(30000)
-    .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', 'browser/4_Ballot_test.sol (BallotTest)')
-    .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', '✓ (Check winning proposal)')
-    .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', '✓ (Check winnin proposal with return value)')
-    .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', 'browser/simple_storage_test.sol (MyTest)')
-    .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', '✓ (Initial value should be100)')
-    .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', '✓ (Value is set200)')
-    .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', 'browser/ks2b_test.sol (kickstarterTest)')
-    .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', '✓ (Check project exists)')
-    .assert.containsText('*[data-id="testTabSolidityUnitTestsOutput"]', '✓ (Check project is fundable)')
-    .end()
+    .pause(5000)
+    .waitForElementPresent('*[data-id="testTabSolidityUnitTestsSummary"]')
+    .assert.containsText('*[data-id="testTabSolidityUnitTestsSummary"]', 'SyntaxError: No visibility specified')
+  },
+
+  'Should fail on deploy': function (browser) {
+    browser.waitForElementPresent('*[data-id="verticalIconsKindfileExplorers"]')
+    .addFile('deployError_test.sol', sources[0]['browser/deployError_test.sol'])
+    .clickLaunchIcon('fileExplorers')
+    .switchFile('browser/deployError_test.sol')
+    .clickLaunchIcon('solidityUnitTesting')
+    .click('*[data-id="testTabCheckAllTests"]')
+    .click('.singleTestLabel:nth-of-type(5)')
+    .scrollAndClick('*[data-id="testTabRunTestsTabRunAction"]')
+    .pause(10000)
+    .waitForElementPresent('*[data-id="testTabSolidityUnitTestsSummary"]')
+    .assert.containsText('*[data-id="testTabSolidityUnitTestsSummary"]', 'contract deployment failed after trying twice')
```

Comment:
I am not sure how this is passing. `deployError_test ` contract will throw revert on execution. Text to check can be `Transaction has been reverted by the EVM`.

`contract deployment failed after trying twice` error is expected in a case where first time contract deployment is failed with OOG error and it also failed when we try it by doubling the gas in next execution.

Reply:
This should be an issue with the plugin for incorrect error thrown, because `contract deployment failed after trying twice` is the error received and i can only make assertions with the displayed output.
![Screenshot from 2020-04-10 10-57-45](https://user-images.githubusercontent.com/23282059/78996374-01fb8c80-7b34-11ea-9617-d05ff445e87d.png)

Reply:
ok, I see

Reply:
This will be further investigated here: https://github.com/ethereum/remix/issues/1453